src/transformers/generation/utils.py

```diff
@@ -1728,10 +1728,19 @@ def _get_initial_cache_position(self, seq_length, device, model_kwargs):
             past_length = 0
             if not isinstance(cache, Cache):
                 past_length = cache[0][0].shape[2]
+                if "ChatGLM" in self.__class__.__name__:
+                    past_length = cache[0][0].shape[0]
```

Comment:
This would be needed because `ChatGLM`, a custom model, also uses a custom cache format.

We don't add logic for custom model's custom choices in `transformers`, my advice would be to open a PR in the Hub repo so as to fix your issue 🤗